### DISCUSSION TITLE
GraphQL: can't get repository by name

### DISCUSSION THREAD

**Discussion by ahaessly** [community/community#24789] [Apps, API and Webhooks]
<p>I can get the name for a repository</p>
<pre><code>{
  "data": {
    "viewer": {
      "repositories": {
        "totalCount": 6,
        "nodes": [
          {
            "name": "jade-data-repo",
            "viewerCanAdminister": true
          }
        ]
      }
    }
  }
}
</code></pre>
<p>but if I try to get that repository by name, it returns null</p>
<pre><code>{
  viewer {
    repository(name: "jade-data-repo") {
      viewerCanAdminister
    }
  }
}

{
  "data": {
    "viewer": {
      "repository": null
    }
  }
}
</code></pre>

**Comment by lee-dohm** [comment, marked as answer]
<p>Thanks for reaching out.</p>
<p>When querying for a repository object, even when you specify <code>viewer</code>, you have to give both the <code>name</code> and <code>owner</code> of the repository you want returned. The reason for this is that I, as a viewer, may have access to both the <code>atom/atom</code> repository and the <code>lee-dohm/atom</code> repository, so specifying <code>atom</code> by itself is ambiguous.</p>
<p>I hope that helps!</p>

**Comment by maxelrod** [comment]
<p>Please provide example query of how this is done</p>
